### PR TITLE
Add release notes for v24.2.0-beta.2

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes REL-1136

### CODE REVIEW

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- [`ALTER BACKUP`]({% link v24.2/alter-backup.md %})
+- [`ALTER CHANGEFEED SET sink`]({% link v24.2/alter-changefeed.md %})
+- [`BACKUP`]({% link v24.2/backup.md %})
+- [`COPY`]({% link v24.2/copy-from.md %})
+- [`CREATE CHANGEFEED`]({% link v24.2/create-changefeed.md %})
+- [`CREATE EXTERNAL CONNECTION`]({% link v24.2/create-external-connection.md %})
+- [`CREATE SCHEDULE FOR BACKUP`]({% link v24.2/create-schedule-for-backup.md %})
+- [`CREATE SCHEDULE FOR CHANGEFEED`]({% link v24.2/create-schedule-for-changefeed.md %})
+- [`EXPORT`]({% link v24.2/export.md %})
+- [`IMPORT`]({% link v24.2/import.md %})
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:20]:
```suggestion
- [`IMPORT`]({% link v24.2/import-into.md %})
```
Should `IMPORT INTO` also be in this list? Should the link text here be updated?

File: src/current/_data/versions.csv [hunk 2]
```diff
@@ -14,4 +14,4 @@ v22.2,2022-12-05,2023-12-05,2024-06-05,N/A,N/A,N/A,N/A,N/A,v22.1,release-22.2
 v23.1,2023-05-15,2024-05-15,2024-11-15,23.1.11,23.1.12,2023-11-13,2024-11-13,2025-11-13,v22.2,release-23.1
 v23.2,2024-02-05,2025-02-05,2025-08-05,23.2.6,23.2.7,2024-07-08,2025-07-08,2026-07-08,v23.1,release-23.2
 v24.1,2024-05-20,2025-05-20,2025-11-20,N/A,N/A,N/A,N/A,N/A,v23.2,release-24.1
-v24.2,N/A,N/A,N/A,N/A,N/A,N/A,N/A,N/A,v24.1,master
+v24.2,N/A,N/A,N/A,N/A,N/A,N/A,N/A,N/A,v24.1,release-24.2
```
Comment (mdlinville) [inline src/current/_data/versions.csv:17]:
```suggestion
v24.2,N/A,N/A,N/A,N/A,N/A,N/A,N/A,N/A,v24.1,master

```
The SQL grammar been branched yet, unfortunately.

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- [`RESTORE`]({% link v24.2/restore.md %})
+- [`SHOW BACKUPS`]({% link v24.2/show-backups.md %})
+- [`SHOW BACKUP`]({% link v24.2/show-backups.md %})
+
+<h3 id="v24-2-0-beta-2-sql-language-changes">SQL language changes</h3>
+
+- Added the [cluster setting]({% link v24.2/cluster-settings.md %}) [`sql.auth.grant_option_for_owner.enabled`](). The default value is `true`, which results in behavior that matches the existing behavior of CockroachDB. When set to `false`, then the [`GRANT OPTION`]({% link v24.2/show-grants.md %}#privilege-grants) is not implicitly given to the owner of an object. The object owner still implicitly has all privileges on the object, just not the ability to grant them to other [users]({% link v24.2/security-reference/authorization#roles.md %}). [#126960][#126960]
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:27]:
```suggestion
- The new [cluster setting]({% link v24.2/cluster-settings.md %}) [`sql.auth.grant_option_for_owner.enabled`]({% link v24.2/cluster-settings.md %}#setting-sql-auth-grant-option-for-owner-enabled) controls whether the owner of an object has permission to grant permission on the object to other users. When set to `true` (the default), the [`GRANT OPTION`]({% link v24.2/show-grants.md %}#privilege-grants) is is implicitly granted to the object owner, who can grant permissions on the object to other [users]({% link v24.2/security-reference/authorization#roles.md %}), preserving the existing behavior of CockroachDB. When set to `false`, the `GRANT OPTION` is not implicitly given to the owner of an object. The owner's permissions do not change, but they can no longer grant permissions to others unless the `GRANT OPTION` is granted to them explicitly.  [#126960][#126960]
```

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+- Fixed a bug in which the `DISCARD` statement was disallowed when the [session setting]({% link v24.2/session-variables.md %}#default-transaction-read-only) `default_transaction_read_only = on`. [#127364][#127364]
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:28]:
```suggestion
- The [`DISCARD`]({% link v24.2/discard.md %}) parameter of [`ALTER TABLE`]({% link v24.2/alter-table.md %}#parameters) can now be used in a read-only transaction when the [session setting]({% link v24.2/session-variables.md %}#default-transaction-read-only) `default_transaction_read_only is set to `on`, This matches the behavior of PostgreSQL. [#127364][#127364] [#127364][#127364]
```

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+<h3 id="v24-2-0-beta-2-bug-fixes">Bug fixes</h3>
+
+- Fixed a bug that could cause [`CREATE INVERTED INDEX`]({% link v24.2/create-index.md %}#create-gin-indexes) and [`ALTER TABLE ... SET LOCALITY REGIONAL BY ROW`]({% link v24.2/set-locality#set-the-table-locality-to-regional-by-row.md %}) statements to fail if the corresponding tables contained columns with non-standard characters in their names such as tabs or newlines. This bug was present since [inverted indexes]({% link v24.2/inverted-indexes.md %}) were introduced in v2.0. [#126996][#126996]
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:32]:
```suggestion
- Fixed a bug that could cause [`CREATE INVERTED INDEX`]({% link v24.2/create-index.md %}#create-gin-indexes) and [`ALTER TABLE ... SET LOCALITY REGIONAL BY ROW`]({% link v24.2/set-locality#set-the-table-locality-to-regional-by-row.md %}) statements to fail if the corresponding table contained columns with non-standard characters in their names, such as tabs or newlines. This bug was introduced along with [inverted indexes]({% link v24.2/inverted-indexes.md %}) in v2.0. [#126996][#126996]
```

**Review by mdlinville — APPROVED** [review]

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md (hunk 1 above)
Comment (rmloveland) [reply src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:20]: based on looking at [this PR](https://github.com/cockroachdb/cockroach/pull/127360/commits/ee740dde8c73b4c07a0d3e4333e3cf032f6d9678) it changes both `IMPORT` and `IMPORT INTO`, so I'll add `IMPORT INTO` to the list

File: src/current/_data/versions.csv (hunk 2 above)
Comment (rmloveland) [reply src/current/_data/versions.csv:17]: ah ok - changed back to 'master'

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md (hunk 3 above)
Comment (rmloveland) [reply src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:27]: thanks this is way better

PS I didn't commit the suggestion via the web UI but I am adding it locally and will be pushing it momentarily

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md (hunk 4 above)
Comment (rmloveland) [reply src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:28]: AFAICT that's not the 'DISCARD' this PR is referring to - the test is here: https://github.com/cockroachdb/cockroach/pull/127298/commits/2c32e2008fcb4cf815dbf28b19d058facdf10a7e

it's this DISCARD which we don't document, not sure why

https://www.postgresql.org/docs/current/sql-discard.html

we have https://cockroachlabs.atlassian.net/browse/DOC-6699 and https://cockroachlabs.atlassian.net/browse/DOC-10475, one of which will require adding some docs for DISCARD, I think

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md (hunk 5 above)
Comment (rmloveland) [reply src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:32]: ✅

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+- Fixed a bug causing [gateway nodes]({% link v24.2/architecture/life-of-a-distributed-transaction.md %}#gateway) to crash while executing [`INSERT`]({% link v24.2/insert.md %}) statements in [`REGIONAL BY ROW`]({% link v24.2/set-locality#set-the-table-locality-to-regional-by-row.md %}) tables. This bug had been present since v23.2. [#127278][#127278]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:33]: nice, thanks

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+- Fixed a bug where [schema changes]({% link v24.2/online-schema-changes.md %}) could hang if the [lease]({% link v24.2/architecture/replication-layer.md %}#leases) [rangefeed]({% link v24.2/create-and-configure-changefeeds.md %}#enable-rangefeeds) stopped getting updates. [#127488][#127488]
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:34]: thanks - went with "stopped receiving"

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+- [`ALTER BACKUP`]({% link v24.2/alter-backup.md %})
+- [`ALTER CHANGEFEED SET sink`]({% link v24.2/alter-changefeed.md %})
+- [`BACKUP`]({% link v24.2/backup.md %})
+- [`COPY`]({% link v24.2/copy-from.md %})
+- [`CREATE CHANGEFEED`]({% link v24.2/create-changefeed.md %})
+- [`CREATE EXTERNAL CONNECTION`]({% link v24.2/create-external-connection.md %})
+- [`CREATE SCHEDULE FOR BACKUP`]({% link v24.2/create-schedule-for-backup.md %})
+- [`CREATE SCHEDULE FOR CHANGEFEED`]({% link v24.2/create-schedule-for-changefeed.md %})
+- [`EXPORT`]({% link v24.2/export.md %})
+- [`IMPORT INTO`]({% link v24.2/import-into.md %})
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:20]: Just verifying that you intended to remove `IMPORT` in addition to adding `IMPORT INTO`?

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+- [`RESTORE`]({% link v24.2/restore.md %})
+- [`SHOW BACKUPS`]({% link v24.2/show-backup.md %})
+- [`SHOW BACKUP`]({% link v24.2/show-backup.md %})
+
+<h3 id="v24-2-0-beta-2-sql-language-changes">SQL language changes</h3>
+
+- The new [cluster setting]({% link v24.2/cluster-settings.md %}) [`sql.auth.grant_option_for_owner.enabled`]({% link v24.2/cluster-settings.md %}#setting-sql-auth-grant-option-for-owner-enabled) controls whether the owner of an object has permission to grant permission on the object to other users. When set to `true` (the default), the [`GRANT OPTION`]({% link v24.2/show-grants.md %}#privilege-grants) is is implicitly granted to the object owner, who can grant permissions on the object to other [users]({% link v24.2/security-reference/authorization.md %}#roles), preserving the existing behavior of CockroachDB. When set to `false`, the `GRANT OPTION` is not implicitly given to the owner of an object. The owner's permissions do not change, but they can no longer grant permissions to others unless the `GRANT OPTION` is granted to them explicitly.  [#126960][#126960]
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:27]:
```suggestion
- The new [cluster setting]({% link v24.2/cluster-settings.md %}) [`sql.auth.grant_option_for_owner.enabled`]({% link v24.2/cluster-settings.md %}#setting-sql-auth-grant-option-for-owner-enabled) controls whether the owner of an object has permission to grant permission on the object to other [users]({% link v24.2/security-reference/authorization.md %}#roles). When set to `true` (the default), the [`GRANT OPTION`]({% link v24.2/show-grants.md %}#privilege-grants) is is implicitly granted to the object owner, who can grant permissions on the object to other users, preserving the existing behavior of CockroachDB. When set to `false`, the `GRANT OPTION` is not implicitly given to the owner of an object. The owner's permissions do not change, but they can no longer grant permissions to others unless the `GRANT OPTION` is granted to them explicitly.  [#126960][#126960]
```

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+- Fixed a bug in which the `DISCARD` statement was disallowed when the [session setting]({% link v24.2/session-variables.md %}#default-transaction-read-only) `default_transaction_read_only = on`. [cockroachdb/cockroach#127364][#127364]
+
+<h3 id="v24-2-0-beta-2-bug-fixes">Bug fixes</h3>
+
+- Fixed a bug that could cause [`CREATE INVERTED INDEX`]({% link v24.2/create-index.md %}#create-gin-indexes) and [`ALTER TABLE ... SET LOCALITY REGIONAL BY ROW`]({% link v24.2/alter-table.md %}#set-the-table-locality-to-regional-by-row) statements to fail if the corresponding table contained columns with non-standard characters in their names such as tabs or newlines. This bug was introduced along with [inverted indexes]({% link v24.2/inverted-indexes.md %}) in v2.0. [#126996][#126996]
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:32]:
```suggestion
- Fixed a bug that could cause [`CREATE INVERTED INDEX`]({% link v24.2/create-index.md %}#create-gin-indexes) and [`ALTER TABLE ... SET LOCALITY REGIONAL BY ROW`]({% link v24.2/alter-table.md %}#set-the-table-locality-to-regional-by-row) statements to fail if the corresponding table contained columns with non-standard characters in their names, such as tabs or newlines. This bug was introduced along with [inverted indexes]({% link v24.2/inverted-indexes.md %}) in v2.0. [#126996][#126996]
```

**Review by mdlinville — APPROVED** [review]
Just a couple trailing nits. LGTM!

File: src/current/_includes/releases/v24.2/v24.2.0-beta.2.md (hunk 8 above)
Comment (rmloveland) [reply src/current/_includes/releases/v24.2/v24.2.0-beta.2.md:20]: apparently there is no more IMPORT page in the v24.2 docs!  TIL

**Pull request merged**